livekit-agents/livekit/agents/job.py

```diff
@@ -99,12 +107,74 @@ async def connect(
 
         await self._room.connect(self._info.url, self._info.token, options=room_options)
         self._on_connect()
+        for p in self._room.remote_participants.values():
+            self._on_participant_connected(p)
 
         _apply_auto_subscribe_opts(self._room, auto_subscribe)
 
     def shutdown(self, reason: str = "") -> None:
         self._on_shutdown(reason)
 
+    def _on_participant_connected(self, p: rtc.RemoteParticipant) -> None:
+        for filter, coro in self._participant_coro_lookup.items():
+            if filter(p):
+                task = asyncio.create_task(coro(p))
+                self._participant_tasks[p.identity] = task
```

Comment:
this confused me a bit in terms of the sequencing (i.e. intention is to run them serially vs override). Does it make sense to enforce serial dispatch for two unrelated callbacks?

imagine this use cases:

```python
async def task1(p: rtc.RemoteParticipant):
    for frame in p.audio_stream:
       pass

async def task2(p: rtc.RemoteParticipant):
    for frame in p.video_stream:
       pass

ctx.add_participant_task(task1)
ctx.add_participant_task(task2)
```

Reply:
Hrmm, I suppose we could actually key off of the (identity,callback) tuple and say that each different callback runs in parallel but only one of each per participant runs at a time.